src/GraphQl/Type/TypeBuilder.php

```diff
@@ -226,4 +205,141 @@ public function isCollection(Type $type): bool
+                }
+
+                return $fields;
+            },
+            'interfaces' => $wrapData ? [] : \array_merge([$this->getNodeInterface()], $interfaceTypes),
+        ];
+
+        return $input ? GraphQLType::nonNull(new InputObjectType($configuration)) : new ObjectType($configuration);
+    }
+
+    private function buildResourceInterfaceType(?string $resourceClass, string $shortName, ResourceMetadata $resourceMetadata, bool $input, ?string $queryName, ?string $mutationName, bool $wrapped, int $depth): ?InterfaceType
+    {
+        static $fieldsBuilder;
+
+        $ioMetadata = $resourceMetadata->getGraphqlAttribute($mutationName ?? $queryName, $input ? 'input' : 'output', null, true);
+        if (null !== $ioMetadata && \array_key_exists('class', $ioMetadata) && null !== $ioMetadata['class']) {
+            $resourceClass = $ioMetadata['class'];
+        }
+
+        $wrapData = !$wrapped && null !== $mutationName && !$input && $depth < 1;
+
+        if ($this->typesContainer->has($shortName)) {
+            $resourceInterface = $this->typesContainer->get($shortName);
+            if (!$resourceInterface instanceof InterfaceType) {
+                throw new \UnexpectedValueException(sprintf('Expected GraphQL type "%s" to be %s.', $shortName, InterfaceType::class));
+            }
+
+            return $resourceInterface;
+        }
+
+        $fieldsBuilder = $fieldsBuilder ?? $this->fieldsBuilderLocator->get('api_platform.graphql.fields_builder');
+
+        $resourceInterface = new InterfaceType([
+            'name' => $shortName,
+            'description' => $resourceMetadata->getDescription(),
+            'fields' => function () use ($resourceClass, $resourceMetadata, $input, $mutationName, $queryName, $wrapData, $depth, $ioMetadata, $fieldsBuilder) {
+                if ($wrapData) {
+                    return [
+                        lcfirst($resourceMetadata->getShortName()) => $this->getResourceObjectType($resourceClass, $resourceMetadata, $input, $queryName, null, true, $depth),
+                    ];
+                }
+
+                return $fieldsBuilder->getResourceObjectTypeFields($resourceClass, $resourceMetadata, $input, $queryName, null, $depth, $ioMetadata);
+            },
+            'resolveType' => function ($value, $context, $info) {
+                if (!isset($value[ItemNormalizer::ITEM_RESOURCE_CLASS_KEY])) {
+                    throw new \UnexpectedValueException('Resource class was not passed. Interface type can not be used.');
+                }
+
+                $shortName = (new \ReflectionClass($value[ItemNormalizer::ITEM_RESOURCE_CLASS_KEY]))->getShortName();
+
+                if (!$this->typesContainer->has($shortName)) {
+                    $shortName .= self::ITEM_POSTFIX;
+                    if (!$this->typesContainer->has($shortName)) {
```

Comment:
This should probably not be used as a fallback, but instead implement the proper checks (i.e. use `Item` postfix, if the serialization context differs between `Item` and `Collection` queries)